src/Fields/TagField.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Ehann\RediSearch\Fields;
+
+class TagField extends AbstractField
+{
+    use Sortable;
+    use Noindex;
+
+    protected $separator = ',';
+
+    public function getType(): string
+    {
+        return 'TAG';
+    }
+
+    public function getSeparator(): string
+    {
+        return $this->separator;
+    }
+
+    public function setSeparator(string $separator)
+    {
+        $this->separator = $separator;
+        return $this;
+    }
+
+    public function getTypeDefinition(): array
+    {
+        $properties = parent::getTypeDefinition();
+
+        $properties[] = 'SEPARATOR';
+        $properties[] = $this->getSeparator();
+
+        if ($this->isSortable()) {
```

Comment:
Are you sure SORTABLE and NOINDEX are supported for tag fields?

Reply:
Sortable is supported, see https://oss.redislabs.com/redisearch/Commands/

> Numeric, tag or text field can have the optional SORTABLE argument that allows the user to later sort the results by the value of this field (this adds memory overhead so do not declare it on large text fields).

And since NOINDEX is used in conjunction with SORTABLE, I guess it's allowed too, at least I don't get an error using it.